Sources_v3/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -241,6 +232,100 @@ open class ChatChannelVC<ExtraData: UIExtraDataTypes>: ViewController,
+                self?.dismiss(animated: true)
+            })
+
+            actions.append(.threadReply { [weak self] in
+                debugPrint("thread reply")
+                self?.dismiss(animated: true)
+            })
+
+            actions.append(.copy { [weak self] in
+                UIPasteboard.general.string = message.text
+                self?.dismiss(animated: true)
+            })
+
+            if message.isSentByCurrentUser {
+                actions.append(.edit { [weak self] in
+                    debugPrint("edit")
+                    self?.dismiss(animated: true)
+                })
+                actions.append(.delete { [weak self] in
+                    debugPrint("delete")
+                    messageController.deleteMessage { [messageController] _ in
+                        self?.dismiss(animated: true)
+                        _ = messageController
+                    }
+                })
+            } else {
+                if currentUser.mutedUsers.contains(message.author) {
+                    actions.append(.unmuteUser { [weak self] in
+                        let userController = messageController.client.userController(userId: message.author.id)
+                        userController.unmute { [userController] _ in
+                            self?.dismiss(animated: true)
+                            _ = userController
+                        }
+                    })
+                } else {
+                    actions.append(.muteUser { [weak self] in
+                        let userController = messageController.client.userController(userId: message.author.id)
+                        userController.mute { [userController] _ in
+                            self?.dismiss(animated: true)
+                            _ = userController
+                        }
+                    })
+                }
+            }
+
+            return actions
+        }
+
+        return .init(
+            view: uiConfig.messageList.messageActionsView.init().withoutAutoresizingMaskConstraints,
+            availableActions: messageActions
+        )
+    }
+
+    private func showActionsPopUp(forMessageIn cell: СhatMessageCollectionViewCell<ExtraData>) {
+        guard let cid = controller.cid, let messageData = cell.message else { return }
+
+        let messageController = controller.client.messageController(
+            cid: cid,
+            messageId: messageData.message.id
+        )
+
+        let popup = ChatMessagePopupViewController<ExtraData>(
+            message: messageData,
+            messageFrameToFit: cell.messageView.superview!.convert(cell.messageView.frame, to: nil),
+            messageActionsController: actionsController(
+                for: messageData,
+                messageController: messageController
+            ),
+            messageReactionsController: .init(
+                view: uiConfig.messageList.messageReactionsView.init().withoutAutoresizingMaskConstraints,
```

Comment:
Why are you injecting view from outside?

Reply:
responded in https://github.com/GetStream/stream-chat-swift/pull/621#discussion_r535892333